### DISCUSSION TITLE
Error: SocketException: Failed host lookup: 'a.bridge.walletconnect.org'

### DISCUSSION THREAD

**Discussion by AnhTu6610** [WalletConnect/walletconnect-monorepo#2856] [Developer Support] [answered]
Error: SocketException: Failed host lookup: 'a.bridge.walletconnect.org' (OS Error: nodename nor servname provided, or not known, errno = 8) Stack Trace: #0 _NativeSocket.startConnect (dart:io-patch/socket_patch.dart:706)

**Comment by chadyj** [comment, marked as answer]
Hey there 👋 
Looks like you are using WalletConnect v1 which has been sunset. You can read more about it here https://medium.com/walletconnect/walletconnect-v1-0-has-now-been-shut-down-ff9baec96690

Please upgrade your project to support v2. Migration details are here https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/overview

**Comment by AnhTu6610** [comment]
I am making an app with functions including connect wallet (Metamask) and send transaction to mint NFT. I can't find a way to do this with this new version.
Who can help me.

**Comment by AnhTu6610** [comment]
I write a function to switchChain BSC testnet, but it only runs to authClient!.request and doesn't have any action on metamask:

  ```
Future switchChain(SessionConnect sessionConnect) async {
    logI(sessionConnect.session.pairingTopic, tag: "pairingTopic");
    launchUrlString("wc:${sessionConnect.session.topic}");
    var req = await authClient!.request(
      topic: sessionConnect.session.topic,
      chainId: "eip155:97",
      request: SessionRequestParams(
        method: "wallet_switchEthereumChain",
        params: [
          {"chainId": chainIdHex},
        ],
      ),
    );
    logI(req, tag: ">>>>>>>>>>>>>>>  req");
  }
```
